src/devices/Common/Iot/Device/Common/AngleExtensions.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using UnitsNet;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="UnitsNet.Angle"/>
+    /// </summary>
+    public static class AngleExtensions
+    {
+        /// <summary>
+        /// Normalizes the angle so it is between 0° and 360° or between -180° and +180° respectively.
+        /// </summary>
+        /// <param name="self">Instance to normalize</param>
+        /// <param name="toFullCircle">Set to true to normalize to 0-360°, otherwise normalizes to +/-180°</param>
+        public static Angle Normalize(this Angle self, bool toFullCircle)
+        {
+            double r = self.Radians;
+            if (toFullCircle)
+            {
+                if (r > Math.PI * 2)
+                {
+                    r = r % (Math.PI * 2);
+                }
+
+                if (r < 0)
+                {
+                    r = -(Math.Abs(r) % (Math.PI * 2));
+                    if (r < 0)
+                    {
+                        r += Math.PI * 2;
+                    }
+                }
+            }
+            else
+            {
+                if (r > Math.PI)
+                {
+                    r = r % (Math.PI * 2);
+                    if (r > Math.PI)
+                    {
+                        // Still above 180?
+                        r -= Math.PI * 2;
+                    }
+                }
+
+                if (r < -Math.PI)
+                {
+                    r = -(Math.Abs(r) % (Math.PI * 2));
+                    if (r < -Math.PI)
+                    {
+                        r += Math.PI * 2;
+                    }
+                }
+            }
+
+            // Return in same unit as original input
+            return Angle.FromRadians(r).ToUnit(self.Unit);
```

Comment:
I'm confused, what's `ToUnit` actually doing? is it just internal representation or something else?

Reply:
The UnitsNet types contain the value in the original unit plus that unit. The calculation within this method needs to use degrees, because we don't know the equivalents for 360° resp 180° for arbitrary units. So at the top of the method, the angle is converted to degrees, and here it is converted back to an Angle instance with the same unit as was initialy used.